src/main.rs

```diff
@@ -1,4 +1,71 @@
-#![deny(clippy::mod_module_files)]
+mod output;
+
 pub mod vector_database;
 
-fn main() {}
+use crate::output::{ChatInterface, TerminalChat};
+
+fn main() {
+    let chat_interface = TerminalChat::new();
+
+    if let Err(e) = chat_interface.start() {
```

Comment:
Alternatively, you could update the main signature to `fn main() -> anyhow::Result<()>` so you could make use of early exit operator
```suggestion
    chat_interface.start()?;
```